JSON/req-schema.json

```diff
@@ -0,0 +1,140 @@
+
+    "type": "object",
+    "properties": {
+        "presets": {
+            "type": "object",
+            "patternProperties": {
+                "^[a-zA-Z0-9_-]*$": {
+                    "type": "array",
+                    "minItems": 1,
+                    "uniqueItems": true,
+                    "items": {
+                        "$ref": "#/definitions/mv_param"
+                    }
+                }
+            },
+            "additionalProperties": false
+        },
+        "validations": {
+            "type": "object",
+            "patternProperties": {
+                "^[A-Za-z]+[A-Za-z0-9_-]*$": {
+                    "$ref": "#/definitions/validation_obj"
+                }
+            },
+            "additionalProperties": false
+        },
+        "units": {
+            "type": "object",
+            "patternProperties": {
+                "^[A-Za-z]+[A-Za-z0-9_-]*$": {
+                    "$ref": "#/definitions/unit_obj"
+                }
+            },
+            "additionalProperties": false
+        }
+    },
+    "required": [ "validations" ],
+    "definitions": {
+        "mv_param": {
+            "type": "object",
+            "properties": {
+                "arg": {
+                    "type": "string",
+                    "minLength": 1
+                },
+                "vals": {
+                    "type": "array",
+                    "minItems": 1,
+                    "uniqueItems": true,
+                    "items": {
+                        "type": "string",
+                        "minLength": 1
+                    }
+                },
+                "role": {
+                    "type": "string",
+                    "enum": ["client", "server", "all"]
+                },
+                "id": {
+                    "type": "string",
+                    "pattern": "^[1-9][0-9]*$"
+                },
+                "enabled": {
+                    "type": "string",
+                    "enum": ["yes", "no"]
+                }
+            },
+            "required": [
+                "arg",
+                "vals"
+            ],
+            "additionalProperties": false
+        },
+        "validation_obj": {
+            "type": "object",
+            "properties": {
+                "args": {
+                    "type": "array",
+                    "minItems": 1,
+                    "uniqueItems": true,
+                    "items": {
+                        "type": "string",
+                        "minLength": 1
+                    }
+                },
+                "vals": {
+                    "type": "string",
+                    "minLength": 1
+                },
+                "description": {
+                    "type": "string",
+                    "minLength": 1
+                },
+                "convert": {
+                    "type": "string",
+                    "minLength": 1
+                },
+                "transform": {
```

Comment:
you need an `"additionalProperties": false` here also...you need it on every object you create if you don't want arbitrary objects to be created by the user.

Reply:
done